### PR TITLE
CGNS-247

### PR DESCRIPTION
**Pull request by MicK7**
Add a context structure for cgio and hdf5 related state

### CODE REVIEW

File: src/pcgnslib.c [hunk 1]
```diff
@@ -35,17 +35,20 @@ freely, subject to the following restrictions:
...
-extern MPI_Comm pcg_mpi_comm;
-extern int pcg_mpi_comm_size;
-extern int pcg_mpi_comm_rank;
-/* Flag indicating if HDF5 file accesses is PARALLEL or NATIVE */
-extern char hdf5_access[64];
-/* flag indicating if mpi_initialized was called */
-extern int pcg_mpi_initialized;
-
-hid_t default_pio_mode = H5FD_MPIO_COLLECTIVE;
+typedef struct _cgns_io_ctx_t {
```
Comment (gsjaardema) [inline src/pcgnslib.c:38]: It looks like we have the same typedef defined both here and in `src/cgns_io.c`.  Shouldn't it be defined once in an include file and included both places?

Comment (MicK7) [reply src/pcgnslib.c:38]: Good idea. It should be better in an header file but the header should be private not to expose this structure outside of cgns.

Comment (gsjaardema) [reply src/pcgnslib.c]: Looks much cleaner.

**Review by gsjaardema — APPROVED** [review]
Looks good to me.  Much cleaner overall

**Review by brtnfld — APPROVED** [review]
LGTM

**Pull request merged**